mantidproject/mantid
# Q1DWeighted SANS improvement

**Description of work.**

Allow Q1DWeighted to treat monochromatic data using multiple samples in the same workspace.
Using each bin as a different sample, for the same wavelength, it is now possible to treat monochromatic data faster without having to go through an external loop. It is a part of an ongoing effort to make SANS kinetic data treatment more efficient.



**To test:**
- Load `ExternalData/Testing/Data/UnitTest/ILL/D11B/kinetic.nxs`
- Run `Q1DWeighted` with rebinning parameters `0, 0.002, 0.1` and default options.
- Check that there are 3 spectra.
- Check that those 3 spectra have curves with actual values.


WIP for #31943 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
